source/metadata.py

```diff
@@ -61,6 +69,12 @@ def read_version():
     if file_path.endswith(".hex") or file_path.endswith(".dfu"):
         # Find out what language this file is
         name: str = os.path.basename(file_path)
+        if ModelName is not None:
+            # If ModelName is provided as the second argument (compatible with make model=NAME fully) but current file name doesn't match the model name, then skip it
+            if not name.startswith(ModelName + "_"):
+                continue
+            if (ModelName == "Pinecil" or ModelName == "Pinecilv2") and not re.match(r"^" + ModelName + "_" + "([A-Z]+).*$", name):
```

Comment:
Would be best to avoid explicitly calling out Pinecil here if we can avoid it

Reply:
I don't like "hard-coding" such kind of stuff myself, believe me. I will think what can be done here but to be fully honest with you I'm not sure that I will find fast & easy way fully avoid it due to tricky variety of "input" file names to parse.